vendor/src/github.com/fluent/fluent-logger-golang/fluent/fluent.go

```diff
@@ -194,7 +210,14 @@ func (f *Fluent) close() (err error) {
 
 // connect establishes a new connection using the specified transport.
 func (f *Fluent) connect() (err error) {
-	f.conn, err = net.DialTimeout("tcp", f.Config.FluentHost+":"+strconv.Itoa(f.Config.FluentPort), f.Config.Timeout)
+	switch f.Config.FluentNetwork {
+	case "tcp":
+		f.conn, err = net.DialTimeout(f.Config.FluentNetwork, f.Config.FluentHost+":"+strconv.Itoa(f.Config.FluentPort), f.Config.Timeout)
```

Comment:
Can we move this host/port vs sock path into a helper function?
This way connect is just a single case and can potentially support other formats easily as well (e.g. "unixgram")